src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = "white";
+    private static final int RADIUS = 10;
+
+    private final Random random = new Random();
+    private final ColorSupplier colorsupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomFigure = random.nextInt(FigureList.values().length);
+        String randomColor = colorsupplier.getRandomColor();
+
+        switch (FigureList.values()[randomFigure]) {
+            case SQUARE:
+                return new Square(randomColor, random.nextInt(6) + 1);
+            case CIRCLE:
+                return new Circle(randomColor,
+                        random.nextInt(8) + 1);
+            case ISOSCELESTRAPEZOID:
+                return new IsoscelesTrapezoid(randomColor,
+                        random.nextInt(10 + 1),
+                        random.nextInt(10 + 1),
+                        random.nextInt(5) + 1);
+            case RECTANGLE:
+                return new Rectangle(randomColor,
```

Comment:
Creating a figure, it's better to use separate variables for the dimensions instead of passing expressions in the constructor for better code readability.